### ISSUE TITLE
Site Editor: 'Delete page' button should be 'Move to trash'

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52639]
### Description

In the Site Editor > Pages > any page, clicking the 'Actions' ellipsis menu reveals a dropdown menu with a 'Delete' button.

<img width="422" alt="Screenshot 2023-07-14 at 14 43 27" src="https://github.com/WordPress/gutenberg/assets/1682452/e5e667c0-1c92-49fd-ab6f-6a401ba33313">

Clicking this button opens a confirm dialog with the text:

`Are you sure you want to delete this page?`

followed by two buttons: Cancel and Delete.

<img width="368" alt="Screenshot 2023-07-14 at 14 44 08" src="https://github.com/WordPress/gutenberg/assets/1682452/a7e08f23-9987-42d0-a5a4-8bf09bd37db7">

Everything in the UI uses the term 'delete'. As a user, I'd think the page is going to be _permanently_ deleted.

Instead, the page gets moved to the Trash. This is a good thing and it's how WordPress always worked. However, the UI is just wrong as it hints to a delete action, while the actual action is moving to trash.

- The text of the button in the dropdown menu should be 'Move to trash' instead of 'Delete'.
- The confirm dialog is redundant. In WordPress, the 'move to trash' action never showed a confirm dialog as it is _not_ a destructive action. Items can be restored from the Trash at any time.

I'm going to add also the Accessibility label because any UI control should always clearly identify the associated action.

### Step-by-step reproduction instructions

- Go to Site Editor > Pages > any page
- In the navigator panel, click the 'Actions ellipsis button, then click 'Delete'.
- A modal dialog opens, asking confirmation. Click Delete.
- The page disappears and the editor navigator panel goes back to the 'Pages' list.
- Go to the WP admin > Pages > Trash at `/wp-admin/edit.php?post_status=trash&post_type=page`
- Observe the 'deleted' page isn't actually deleted. It's there in the Trash.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
More wording needs to be changed, including the snackbar notice:

<img width="199" alt="Screenshot 2023-07-14 at 15 25 24" src="https://github.com/WordPress/gutenberg/assets/1682452/b71cdc61-835c-4995-a755-43f74917b415">
